lib/commands/connect/auth_connection.js

```diff
@@ -27,27 +29,24 @@ module.exports = {
     ],
     needsApp: true,
     needsAuth: true,
-    run: cli.command(function (context, heroku) {
-      return api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku)
-        .then(function(connections) {
-          if (connections.length > 0) {
-            callbackServer();
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withConnection(context, heroku);
+      callbackServer();
+
+      let url = '/api/v3/connections/' + connection.id + '/authorize_url';
+      let args = {'environment':'production', 'next': 'http://127.0.0.1:18000'};
+      if (context.flags.login) {
+        args.login_url = context.flags.login;
+      }
+
+      let response = yield api.request(context.auth.password, 'POST', url, args);
+      let redir = response.json.redirect;
 
-            try {
-              let args = {'environment':'production', 'next': 'http://127.0.0.1:18000'};
-              if (context.flags.login) {
-                args.login_url = context.flags.login;
-              }
-              api.request(context.auth.password,
-                'POST', '/api/v3/connections/' + connections[0].id + '/authorize_url', args).then(function(response) {
-                  let redir = response.json.redirect;
-                  console.log("Launching Salesforce for authorization. If your browser doesn't open, please copy the following URL to proceed:\n\n" + redir + "\n");
-                  open(redir);
-              });
-            } catch (e) {console.log(e)};
-          } else {
-            console.log("No connection found");
-          }
-        });
-    })
+      console.log("Launching Salesforce for authorization. If your browser doesn't open, please copy the following URL to proceed:\n\n" + redir + "\n");
+      try {
+        open(redir);
```

Comment:
Ah, yes. I had actually prepped another PR to do just that, before remembering that I already had this one in progress. I'll update this to use it, but I'll hold off on removing the depdendency until we've resolved #26, which also uses `open()`.